src/main/scripts/install.sh

```diff
@@ -0,0 +1,215 @@
+    fi
+
+    echo "cluster \"${clusterName}\" is successfully created!"
+}
+
+create_custom_profile() {
+    profileName=$1
+    hostName=$2
+    nodeName=$3
+    dmgrHostName=$4
+    dmgrPort=$5
+    dmgrAdminUserName=$6
+    dmgrAdminPassword=$7
+    
+    curl $dmgrHostName:$dmgrPort --output - >/dev/null 2>&1
+    while [ $? -ne 56 ]
+    do
+        sleep 5
+        echo "dmgr is not ready"
+        curl $dmgrHostName:$dmgrPort --output - >/dev/null 2>&1
+    done
+    sleep 60
+    echo "dmgr is ready to add nodes"
+
+    output=$(${WAS_ND_INSTALL_DIRECTORY}/bin/manageprofiles.sh -create -profileName $profileName -hostName $hostName -nodeName $nodeName \
+        -profilePath ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName -templatePath ${WAS_ND_INSTALL_DIRECTORY}/profileTemplates/managed \
+        -dmgrHost $dmgrHostName -dmgrPort $dmgrPort -dmgrAdminUserName $dmgrAdminUserName -dmgrAdminPassword $dmgrAdminPassword 2>&1)
+    while echo $output | grep -qv "SUCCESS"
+    do
+        sleep 10
+        echo "adding node failed, retry it later..."
+        rm -rf ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName
+        output=$(${WAS_ND_INSTALL_DIRECTORY}/bin/manageprofiles.sh -create -profileName $profileName -hostName $hostName -nodeName $nodeName \
+            -profilePath ${WAS_ND_INSTALL_DIRECTORY}/profiles/$profileName -templatePath ${WAS_ND_INSTALL_DIRECTORY}/profileTemplates/managed \
+            -dmgrHost $dmgrHostName -dmgrPort $dmgrPort -dmgrAdminUserName $dmgrAdminUserName -dmgrAdminPassword $dmgrAdminPassword 2>&1)
+    done
+    echo $output
+}
+
+while getopts "m:c:f:h:r:x:" opt; do
+    case $opt in
+        m)
+            adminUserName=$OPTARG #User id for admimistrating WebSphere Admin Console
+        ;;
+        c)
+            adminPassword=$OPTARG #Password for administrating WebSphere Admin Console
+        ;;
+        f)
+            dmgr=$OPTARG #Flag indicating whether to install deployment manager
+        ;;
+        h)
+            dmgrHostName=$OPTARG #Host name of deployment manager server
+        ;;
+        r)
+            members=$OPTARG #Number of cluster members
+        ;;
+        x)
+            dynamic=$OPTARG #Flag indicating whether to create a dynamic cluster or not
+        ;;
+    esac
+done
+
+# Check whether the user is entitled or not
+while [ ! -f "/var/log/cloud-init-was.log" ]
+do
+    sleep 5
+done
+
+isDone=false
+while [ $isDone = false ]
+do
+    result=`(tail -n1) </var/log/cloud-init-was.log`
+    if [[ $result = Unentitled ]] || [[ $result = Entitled ]]; then
+        isDone=true
+    else
+        sleep 5
+    fi
+done
+echo "The input IBMid account is ${result}."
+
+# Remove cloud-init artifacts and logs
+cloud-init clean --logs
+
+# Terminate the process for the un-entitled user
+if [ ${result} = Unentitled ]; then
+    exit 1
+fi
+
+# Update applications installed on the system
+yum update -y
```

Comment:
If we do yum update, it will add more time to overall process depending on how many updates are pending. Also, if yum updates required a reboot then will it interrupt the whole process?

Reply:
Good question. Since `yum update -y` is already done when provisioning the VM for base image (see this [code snippet](https://github.com/WASdev/azure.websphere-traditional.image/blob/master/src/main/scripts/install.sh#L40)), I guess it can be removed here?

Reply:
`yum update -y` is removed in [this commit](https://github.com/WASdev/azure.websphere-traditional.cluster/pull/9/commits/8b1757808a56d80b30149c078b6a25c304688343).

Reply:
Hello @git4rk , our workflow model has **yum update** as a process that happens at image creation time, automated by the pipeline. As you said, it will some time to do it at deployment time, so let's just leave it at image creation time.